Snow System(Proper) & Client /Roll system

SargeDG opened the pull request:
Snow system allows snow to be enabled/disabled via script /snow <true/false>
script sends all online players the snow packet so online players do not need to relog
script writes the valid user input into WorldManager.Instance.IsSnowing
script announces to server that Snow was Enabled/Disabled


created a public bool "IsSnowing" in WorldManager
created method OnPlayerJoin() in WorldManager
created method Snow() in WorldManager
created a call to WorldManager.OnplayerJoin() in CSNotifyInGameCompletedPacket.cs
Created script Snow.cs

AAGene commented on AAEmu.Game/Scripts/Commands/Snow.cs:
```diff
@@ -0,0 +1,50 @@
+                WorldManager.Instance.BroadcastPacketToServer(new SCOnOffSnowPacket(isSnowing));
+
+                //announce to all players on the server snow was enabled or disabled
+                WorldManager.Instance.BroadcastPacketToServer(new SCNoticeMessagePacket(3, "9#0823d1", 3000, "  Snow Was set to " + isSnowing.ToString()));
```
This should not be sent to every player. Have you ever seen such a message when playing ArcheAge ? ;)

AAGene commented on AAEmu.Game/Core/Managers/World/WorldManager.cs:
```diff
@@ -496,5 +497,31 @@ private bool ValidRegion(uint worldId, int x, int y)
+            }
+
+
+
```
3 empty lines ? why ?

AAGene commented on AAEmu.Game/Core/Managers/World/WorldManager.cs:
```diff
@@ -496,5 +497,31 @@ private bool ValidRegion(uint worldId, int x, int y)
+
+            //turn snow on off 
+            Snow(IsSnowing, character);
+            _log.Warn("Snow value sent to " + character.Name);
```
Useless warn

AAGene commented on AAEmu.Game/Core/Managers/World/WorldManager.cs:
```diff
@@ -496,5 +497,31 @@ private bool ValidRegion(uint worldId, int x, int y)
+        public void Snow(bool onoff, Character character)
+        {
+            // set snowing state to value of onoff
+            Instance.IsSnowing = onoff;
```
bool onoff -> useless

The only call to "Snow" you do uses "IsSnowing". You're essentially doing 

Instance.IsSnowing = Instance.IsSnowing

SargeDG commented:
Fixed all The Issues that were presented, also added another commit 

[new commit: Client /Roll]
Completed Server side Functionality for clients baked in /roll chat command

Created method PlayerRoll(character self, int min, int max) in CharacterManager
created call to PlayerRoll() from CSRollDicePacket

AAGene commented on AAEmu.Game/Core/Managers/UnitManagers/CharacterManager.cs:
```diff
@@ -381,6 +381,14 @@ public void Load()
+        {
+
+            var roll = Rand.Next(min, max);
+            Self.SendMessage(roll.ToString() + ".");
```
Multiple issues here : 

- The roll should be sent to nearby players too. When rolling for loot, usually you want to see other people's rolls too.
- Pretty sure there is either a packet, or a localized string for this.
- Just the number ? What about "'X rolled Y."

SargeDG replied:
ok so would this be more approriate?
Self.BroadcastPacket(new SCChatMessagePacket(ChatType.System, string.Format(Self.Name + "Rolled " + roll.ToString()  + ".")), true);

SargeDG replied:
if you are referring to SCChatLocalizedMessagePacket the only part of it that exsits in in the project is the offset for the packet '0xd2' in SCOffsets.cs

zarobhr replied:
just my 2cents, i believe that if you are not in a group then sending to nearby players is fine, but if you are in a group/raid it should only be send to raid/group members and Not nearby players

SargeDG replied:
going to get down with SQLite and find or create the localized text for it

ZeromusXYZ replied:
In case you want to do it proper ...
The text for /roll is at `ui_texts text idx = 2412`
the resulting text is `|cFFFF6600 $1 rolled a $2 on the dice. (1-$3)`

SargeDG replied:
yes i did find it in the db the issue im having is pulling that into memory as the localization manager NLObP gave me only loads 94 of the 316708 rows in compact.sqlite table localized_texts

AAGene commented on AAEmu.Game/Core/Packets/C2G/CSRollDicePacket.cs:
```diff
@@ -11,9 +14,11 @@ public CSRollDicePacket() : base(0x0bd, 1)
             var max = stream.ReadUInt32();
-            
-            _log.Warn("RollDice, Max: {0}", max);
+            int min = 1;            
```
Why a min ? If you're going to set a default value, don't bother passing the min in your function, just hardcode it.

SargeDG replied:
just wanted it to  be easy to read i can change it

zarobhr replied:
if the roll is 1 to 100, you want 100 possibilities , but the rand function is exclusive of maxvalue so you will only get a value of 1 to 99
you be setting max to max +1 somewhere so its  1 to 101 , to get 100 possibilities
simpler  1 to 6 will never roll a 6 but 1 to 7 will

SargeDG replied:
Thanks, ill implement a fix shortly
